File: src/main/java/org/fountainmc/api/world/tileentity/Jukebox.java
package org.fountainmc.api.world.tileentity;

import javax.annotation.Nullable;

import org.fountainmc.api.inventory.item.Item;
import org.fountainmc.api.world.block.JukeboxState;

public interface Jukebox extends TileEntity {
    @Nullable
    Item getRecord();

    default boolean hasRecord() {
        return getRecord() != null;
    }

    @Override
    default JukeboxState getBlockState() {
        return (JukeboxState) TileEntity.super.getBlockState();
    }
}

File: src/main/kotlin/org/fountainmc/api/world/tileentity/Jukebox.kt
package org.fountainmc.api.world.tileentity

import org.fountainmc.api.inventory.item.Item
import org.fountainmc.api.world.block.JukeboxState

interface Jukebox : TileEntity<JukeboxState> {
    val record: Item?

    val hasRecord
        get() = record != null
}
